src/client/sky.cpp

```diff
@@ -700,69 +570,198 @@ void Sky::update(float time_of_day, float time_brightness,
+			pointcolor = m_mix_scolor(pointcolor_moon, pointcolor_sun,
+					pointcolor_blend);
 		}
 		m_bgcolor = m_mix_scolor(m_bgcolor, pointcolor, m_horizon_blend() * 0.5);
-		m_skycolor = m_mix_scolor(m_skycolor, pointcolor, m_horizon_blend() * 0.25);
+		m_skycolor = m_mix_scolor(
+				m_skycolor, pointcolor, m_horizon_blend() * 0.25);
 	}
 
 	float cloud_direct_brightness = 0.0f;
 	if (sunlight_seen) {
 		if (!m_directional_colored_fog) {
 			cloud_direct_brightness = time_brightness;
-			// Boost cloud brightness relative to sky, at dawn, dusk and at night
+			// Boost cloud brightness relative to sky, at dawn, dusk and at
+			// night
 			if (time_brightness < 0.7f)
 				cloud_direct_brightness *= 1.3f;
 		} else {
-			cloud_direct_brightness = std::fmin(m_horizon_blend() * 0.15f +
-				m_time_brightness, 1.0f);
+			cloud_direct_brightness = std::fmin(
+					m_horizon_blend() * 0.15f + m_time_brightness,
+					1.0f);
 			// Set the same minimum cloud brightness at night
 			if (time_brightness < 0.5f)
-				cloud_direct_brightness = std::fmax(cloud_direct_brightness,
-					time_brightness * 1.3f);
+				cloud_direct_brightness =
+						std::fmax(cloud_direct_brightness,
+								time_brightness * 1.3f);
 		}
 	} else {
 		cloud_direct_brightness = direct_brightness;
 	}
 
-	m_cloud_brightness = m_cloud_brightness * cloud_color_change_fraction +
-		cloud_direct_brightness * (1.0 - cloud_color_change_fraction);
-	m_cloudcolor_f = video::SColorf(
-		m_cloudcolor_bright_f.r * m_cloud_brightness,
-		m_cloudcolor_bright_f.g * m_cloud_brightness,
-		m_cloudcolor_bright_f.b * m_cloud_brightness,
-		1.0
-	);
+	m_cloud_brightness =
+			m_cloud_brightness * cloud_color_change_fraction +
+			cloud_direct_brightness * (1.0 - cloud_color_change_fraction);
+	m_cloudcolor_f = video::SColorf(m_cloudcolor_bright_f.r * m_cloud_brightness,
+			m_cloudcolor_bright_f.g * m_cloud_brightness,
+			m_cloudcolor_bright_f.b * m_cloud_brightness, 1.0);
 	if (m_directional_colored_fog) {
-		m_cloudcolor_f = m_mix_scolorf(m_cloudcolor_f,
-			video::SColorf(pointcolor), m_horizon_blend() * 0.25);
+		m_cloudcolor_f = m_mix_scolorf(m_cloudcolor_f, video::SColorf(pointcolor),
+				m_horizon_blend() * 0.25);
+	}
+}
+
+void Sky::draw_sun(video::IVideoDriver *driver, float sunsize, video::SColor suncolor,
+		video::SColor suncolor2, float wicked_time_of_day)
+{
+	static const u16 indices[4] = {0, 1, 2, 3};
+	std::array<video::S3DVertex, 4> vertices;
+	if (!m_sun_texture) {
+		driver->setMaterial(m_materials[1]);
+		const float sunsizes[4] = {
+				sunsize * 1.7f, sunsize * 1.2f, sunsize, sunsize * 0.7f};
+		video::SColor c1 = suncolor;
+		video::SColor c2 = suncolor;
+		c1.setAlpha(0.05 * 255);
+		c2.setAlpha(0.15 * 255);
+		const video::SColor colors[4] = {c1, c2, suncolor, suncolor2};
+		for (int i = 0; i < 4; i++) {
+			vertices = draw_sky_body(-sunsizes[i], sunsizes[i], colors[i]);
+			vertices = place_sky_body(
+					vertices, 90, wicked_time_of_day * 360 - 90);
```

Comment:
Please only wrap lines at 90 columns (tab size 4). This may be a bad clang format suggestion.